### PR TITLE
prefix autoupdater error with statuscode and domain

### PR DESCRIPTION
**Pull request by codebytere**
Fixes https://github.com/electron/electron/issues/7868

Prepends NSError `code` and `domain`  to `localizedErrorDescription` such that the value seen on error is `code:domain:localizedError Description`.

Edited File: `electron/atom/browser/auto_updater_mac.mm`

### CODE REVIEW

File: atom/browser/ui/file_dialog_kde.cc [hunk 1]
```diff
@@ -0,0 +1,63 @@
+#include "atom/browser/ui/file_dialog.h"
```
Comment (MarshallOfSound) [inline atom/browser/ui/file_dialog_kde.cc]: I think this file was checked in by accident?

File: atom/browser/auto_updater_mac.mm [hunk 2]
```diff
@@ -104,8 +104,8 @@
           delegate->OnUpdateNotAvailable();
         }
       } error:^(NSError *error) {
-        NSMutableString* failureString =
-          [NSMutableString stringWithString:error.localizedDescription];
+        NSMutableString *failureString =
+          [[NSString stringWithFormat:@"%@:%@:%@", error.code, error.domain, error.localizedDescription] mutableCopy];
```
Comment (MarshallOfSound) [inline atom/browser/auto_updater_mac.mm]: Thinking whether or not this is a breaking change if people were previously using this to determine the error.  Might be a safer solution to set `code` and `domain` as attributes on the emitted error object.

You can do that pretty easily in `atom_api_auto_updater.cc`.  Just need to change the `OnError` method to accept a code and a message as well.

```cpp
void AutoUpdater::OnError(const std::string& message, const int code, const std::string& domain) {
  v8::Locker locker(isolate());
  v8::HandleScope handle_scope(isolate());
  auto error = v8::Exception::Error(mate::StringToV8(isolate(), message));
  auto errorObject = error->ToObject(isolate()->GetCurrentContext()).ToLocalChecked();
  errorObject->Set(mate::StringToV8(isolate(), "code"), v8::Integer::New(isolate(), code));
  errorObject->Set(mate::StringToV8(isolate(), "domain"), mate::StringToV8(isolate(), domain));
  mate::EmitEvent(
      isolate(),
      GetWrapper(),
      "error",
      errorObject,
      // Message is also emitted to keep compatibility with old code.
      message);
}
```

File: atom/browser/ui/file_dialog_kde.cc (hunk 1 above)
Comment (codebytere) [inline atom/browser/ui/file_dialog_kde.cc]: it was, changing now!

File: atom/browser/api/atom_api_auto_updater.h [hunk 3]
```diff
@@ -32,6 +32,8 @@ class AutoUpdater : public mate::EventEmitter<AutoUpdater>,
 
   // Delegate implementations.
   void OnError(const std::string& error) override;
+  void OnError(const std::string& message,
+    const int code, const std::string& domain);
```
Comment (MarshallOfSound) [inline atom/browser/api/atom_api_auto_updater.h]: normally we try to line up this line with the start of the params on the line above, thought that was actually a linting rule

**Comment by MarshallOfSound** [comment]
🚢

**Pull request merged**